dashboard/src/modules/components/DatePickerComponent/index.jsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState } from "react";
+import {
+  InputGroup,
+  InputGroupText,
+  DatePicker,
+  isValidDate,
+  yyyyMMddFormat,
+  Button,
+} from "@patternfly/react-core";
+import moment from "moment";
+
+function DatePickerWidget({
+  dataArray,
+  setPublicData,
+  controllerName,
+  setDateRange,
+}) {
+  const [fromDate, setFromDate] = useState(
+    moment(new Date(1990, 10, 4)).format("YYYY/MM/DD")
+  );
+  const [toDate, setToDate] = useState(
+    moment(new Date(2040, 10, 4)).format("YYYY/MM/DD")
+  );
+  const toValidator = (date) =>
+    isValidDate(fromDate) && date >= fromDate
+      ? ""
+      : "To date must be less than from date";
+  let modifiedArray = [];
+  const onFromChange = (_str, date) => {
+    setFromDate(new Date(date));
+    moment(date).format("DD/MM/YYYY");
+    if (isValidDate(date)) {
+      date.setDate(date.getDate() + 1);
+      setToDate(yyyyMMddFormat(date));
+    } else {
+      setToDate("");
+    }
+  };
+
+  const filterByDate = () => {
+    modifiedArray = dataArray.filter((data) => {
+      let formattedData = moment(data.metadata["dataset.created"]).format(
+        "YYYY/MM/DD"
+      );
+      return (
+        Date.parse(formattedData) >= Date.parse(fromDate) &&
+        Date.parse(formattedData) <= Date.parse(toDate) &&
+        data.name.includes(controllerName)
+      );
+    });
+    setPublicData(modifiedArray);
+    setDateRange(fromDate, toDate);
+  };
+  return (
+    <InputGroup style={{ marginLeft: "10px" }}>
+      <InputGroupText>Filter By Date</InputGroupText>
+      <DatePicker
+        onChange={onFromChange}
+        aria-label="Start date"
+        placeholder="YYYY-MM-DD"
+      />
+      <InputGroupText>to</InputGroupText>
+      <DatePicker
+        value={toDate}
+        onChange={(date) => setToDate(date)}
+        isDisabled={!isValidDate(fromDate)}
+        rangeStart={fromDate}
+        validators={[toValidator]}
+        aria-label="End date"
+        placeholder="YYYY-MM-DD"
+      />
```

Comment:
few extra attributes are required in the toDate Date Picker,like rangeStart attribute disable the dates before the selected fromDate

Reply:
> `rangeStart` attribute disable the dates before the selected `fromDate`

That makes sense.  (I wonder if we should limit the dates available to `fromDate`?...for instance, if we know what the earliest data set date is, we could disable dates prior to that.  It might be more effort than it is worth, and it could get us into trouble, but it might be nice.)

But, what about `value`, `validators`, and perhaps `isDisabled`?

Reply:
isDisabled attribute will disable toDate date picker if the fromDate is an invalid Date (like an ill formatted one or something like 2022-03-34 (which subsequently don’t exist))

The validator attribute returns an error message if the date entered is an invalid one

The value attribute is to provide an initial value to date Picker when the page is loaded (in this case it will be Present day date)

Reply:
> isDisabled attribute will disable toDate date picker if the fromDate is an invalid Date (like an ill formatted one or something like 2022-03-34 (which subsequently don’t exist))

😎  

> The validator attribute returns an error message if the date entered is an invalid one

I'm guessing that this is an overrider and that `fromDate` is using Paternfly's/`<DatePicker>`'s default validator...is that right?

> The value attribute is to provide an initial value to date Picker when the page is loaded (in this case it will be Present day date)

I think I'm starting to understand the flow, here.  Looking at the [Patternfly sample code](https://www.patternfly.org/v4/components/date-picker/react-demos/), when the page is first loaded, the pickers display the placeholder text (because they have no other values), and the "to" picker is disabled until a (mostly) valid value is entered via the "from" picker (I say "mostly valid", because, if you type it in manually, you can get it to accept 29-31 February, even in non-leapyears...); once the "from" date has been picked, then the "to" picker's placeholder is replaced by a valid date calculated from the "from" date and the picker is enabled.

So, the "to" picker needs the `value` attribute, so that the "from" picker can set it.  The "from" picker never needs to be disabled because it's always available to be set (i.e., it's supposed to be set first).  Similarly, the "from" picker doesn't need a custom validator, because it is supposed to be set first and is not constrained by the "to" value.

The point is, the pickers are meant to be cascading and they are not meant to be symmetrical, so they necessarily have different attributes.  (I'm not really a fan of the cascading model...while it makes the code behind it simpler, perhaps, it puts constraints on the user which might be counterintuitive and frustrating, but we can let that go.)